### DISCUSSION TITLE
Issue with creating custom chain.

### DISCUSSION THREAD

**Discussion by randomdevver** [rainbow-me/rainbowkit#1386] [Q&A] [answered]
Hello i am trying to add a custom chain via https://www.rainbowkit.com/docs/custom-chains, but it seems like it is outdated,

getting these issues: 
`Property 'public' is missing in type '{ default: { http: string[]; }; }' but required in type '{ [key: string]: RpcUrls; default: RpcUrls; public: RpcUrls; }'.ts(2741)
index.d.ts(19, 9): 'public' is declared here.
index.d.ts(16, 5): The expected type comes from property 'rpcUrls' which is declared here on type 'Chain'`

`Property 'provider' does not exist on type '{ readonly chains: Chain[]; readonly publicClient: ({ chainId }: { chainId?: number | undefined; }) => PublicClient<FallbackTransport>; readonly webSocketPublicClient: ({ chainId }: { ...; }) => WebSocketPublicClient<...> | undefined; }'.`

any help is appreciated, ( am looking to add op bnb testnet)

**Comment by DanielSinclair** [comment, marked as answer]
Oops, you're right that the docs are outdated. Opened a PR to fix that. You only need to make a small change to your Custom Chain by adding `public` under `rpcUrls`. Reference the updated Wagmi example below:
```ts
import { Chain } from 'wagmi'
 
const avalanche: Chain = {
  id: 43_114,
  name: 'Avalanche',
  network: 'avalanche',
  iconUrl: 'https://example.com/icon.svg',
  iconBackground: '#fff',
  nativeCurrency: {
    decimals: 18,
    name: 'Avalanche',
    symbol: 'AVAX',
  },
  rpcUrls: {
    public: { http: ['https://api.avax.network/ext/bc/C/rpc'] },
    default: { http: ['https://api.avax.network/ext/bc/C/rpc'] },
  },
  blockExplorers: {
    default: { name: 'SnowTrace', url: 'https://snowtrace.io' },
    etherscan: { name: 'SnowTrace', url: 'https://snowtrace.io' },
  },
  contracts: {
    multicall3: {
      address: '0xca11bde05977b3631167028862be2a173976ca11',
      blockCreated: 11_907_934,
    },
  },
  testnet: false,
};
```

**Reply by randomdevver** [reply]
Thank you Daniel
